packages/lib-kaspa/index.ts

```diff
@@ -0,0 +1,169 @@
+        return {
+          previousOutpoint: {
+            transactionId: previousOutpoint.transactionId,
+            index: previousOutpoint.index,
+          },
+          signatureScript: input.signatureScript,
+          sequence: Number(input.sequence),
+          sigOpCount: input.sigOpCount,
+        }
+      }),
+      outputs: transaction.outputs.map((output: TransactionOutput) => ({
+        amount: Number(output.value),
+        scriptPublicKey: {
+          version: 0,
+          scriptPublicKey: output.scriptPublicKey.script,
+        },
+      })),
+      lockTime: Number(transaction.lock_time),
+      subnetworkId: transaction.subnetworkId,
+    },
+    allowOrphan: true,
+  }
+}
+
+const stripHexPrefix = (hex: string): string => {
+  return hex.replace(/^0x/, '')
+}
+
+const assertSigned = (res: GenerateSignatureResponse) => {
+  if (res.status === 'Failed') {
+    throw new DfnsError(-1, 'signing failed', res)
+  } else if (res.status !== 'Signed') {
+    throw new DfnsError(
+      -1,
+      'cannot complete signing synchronously because this wallet action requires policy approval',
+      res
+    )
+  }
+}
+
+export class DfnsWallet {
+  private readonly dfnsClient: DfnsApiClient
+
+  private constructor(private metadata: WalletMetadata, options: DfnsWalletOptions) {
+    this.dfnsClient = options.dfnsClient
+  }
+
+  public static async init(options: DfnsWalletOptions) {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    if (res.network !== 'Kaspa') {
+      throw new DfnsError(-1, 'wallet is not bound to Kaspa', {
+        walletId,
+        network: res.network,
+      })
+    }
+
+    return new DfnsWallet(res, options)
+  }
+
+  public get address(): string {
+    return this.metadata.address!
+  }
+
+  public get publicKey(): string {
+    return this.metadata.signingKey.publicKey!
+  }
+
+  public decodeHexTransaction(tx: string): Transaction {
+    return Transaction.deserializeFromJSON(Buffer.from(stripHexPrefix(tx), 'hex').toString())
+  }
+  
+  public encodeTransaction(tx: Transaction): string {
+    return Buffer.from(tx.serializeToJSON()).toString('hex')
+  }
+
+  public async sign(transaction: Transaction): Promise<Transaction> {
+    const res = await this.dfnsClient.wallets.generateSignature({
+      walletId: this.metadata.id,
+      body: {
+        kind: 'Transaction',
+        transaction: `0x${this.encodeTransaction(transaction)}`,
+      },
+    })
+
+    assertSigned(res)
+    if (!res.signedData) {
+      throw new DfnsError(-1, 'signature missing', res)
+    }
+
+    return this.decodeHexTransaction(res.signedData)
+  }
+
+  public buildTransaction(utxos: any[], to: string, amount: bigint, feePerUtxo: bigint): Transaction {
```

Comment:
I rather not be responsible for this. should be similar to bitcoinjs integration. let developers be responsible for constructing the correct transaction however they want, and we will just sign it and return signed transaction back. and also expose a `signHash` method.